configstack/running_module.go

```diff
@@ -181,8 +182,12 @@ func collectErrors(modules map[string]*runningModule) error {
 }
 
 // Run a module once all of its dependencies have finished executing.
-func (module *runningModule) runModuleWhenReady() {
+func (module *runningModule) runModuleWhenReady(s chan struct{}) {
 	err := module.waitForDependencies()
+	s <- struct{}{} // Lock
+	defer func() {
+		<-s // Unlock
+	}()
```

Comment:
It's not really lock and unlock right... It's add 1 to the semaphore and subtract 1? Perhaps add an explanation here that writing to the channel will block if max concurrency is exceeded.